### PR TITLE
Update the dependencies and remove the BC layers

### PR DESCRIPTION
**Pull request by leofeyer**
* [x] Make PHP 7.4 the minimum PHP version
* [x] Remove old minor versions that were still allowed due to BC
* [x] Remove the BC layers
* [x] Upgrade to Symfony 5.3 only (will be changed later to Symfony 5.4 only)
* [x] Use the current major versions where possible
* [ ] Use `doctrine/dbal` in version 3

### CODE REVIEW

**Comment by m-vo** [comment]
There are also still some conditional statements checking for `PHP_VERSION_ID` in the codebase that we can drop.

Should we also refactor code? With PHP7.4 we could finally use…
* typed properties
* arrow functions (`fn(…) => …`)  without the need for `use` 
* shorter null coalescing assignment (`$a ??= $b`)
* array unpacking (`['a', 'b', ...$items]`) instead of `array_merge`

**Comment by leofeyer** [comment]
Easy, tiger, this PR is still a draft. 😄

**Comment by m-vo** [comment]
It's the excitement :smile:

**Comment by leofeyer** [comment]
I feel you. 😄

**Comment by leofeyer** [comment]
Since the PR is already pretty huge, let's merge it as is and add the other changes (doctrine/dbal v3, PHP 7.4 features etc.) later in separate PRs.

File: core-bundle/src/Controller/BackendController.php [hunk 1]
```diff
@@ -182,12 +181,7 @@ public function pickerAction(Request $request): RedirectResponse
...
         if ($request->query->has('extras')) {
-            if ($request->query instanceof InputBag) {
-                $extras = $request->query->all('extras');
-            } else { /** @phpstan-ignore-line */
-                // Backwards compatibility with symfony/http-foundation <5.0
-                $extras = $request->query->get('extras');
-            }
+            $extras = $request->query->all('extras');
 
             if (empty($extras) || !\is_array($extras)) {
```
Comment (m-vo) [inline core-bundle/src/Controller/BackendController.php]: `InputBag#all()` always returns an array, so we can IMHO drop the `is_array` check.

**Review by bytehead — DISMISSED** [review]
LGTM.

**Review by m-vo — CHANGES_REQUESTED** [review]

File: core-bundle/src/Controller/BackendController.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Controller/BackendController.php]: Removed in ad0ef7b3b5a99aea231805324931b5da23f59315.

**Review by m-vo — APPROVED** [review]

**Review by bytehead — APPROVED** [review]

**Pull request merged**